connectors/sources/salesforce.py

```diff
@@ -0,0 +1,1443 @@
+        all_case_feeds = []
+        async for case_feeds in self._yield_non_bulk_query_pages(query):
+            all_case_feeds.extend(case_feeds)
+
+        return all_case_feeds
+
+    def _parse_content_documents(self, record):
+        content_docs = []
+        content_links = record.get("ContentDocumentLinks", {}) or {}
+        content_links = content_links.get("records", []) or []
+        for content_link in content_links:
+            content_doc = content_link.get("ContentDocument")
+            if not content_doc:
+                continue
+
+            content_doc["linked_sobject_id"] = record.get("Id")
+            content_docs.append(content_doc)
+
+        return content_docs
+
+    async def download_content_documents(self, content_documents):
+        for content_document in content_documents:
+            content_version = content_document.get("LatestPublishedVersion", {}) or {}
+            download_url = content_version.get("VersionDataUrl")
+            if download_url:
+                content_document["_attachment"] = await self._download(download_url)
+
+            yield content_document
+
+    async def queryable_sobjects(self):
+        """Cached async property"""
+        if self._queryable_sobjects is not None:
+            return self._queryable_sobjects
+
+        response = await self._get_json(f"{self.base_url}{DESCRIBE_ENDPOINT}")
+        self._queryable_sobjects = []
+
+        for sobject in response.get("sobjects", []):
+            if sobject["queryable"] is True and sobject["name"] in RELEVANT_SOBJECTS:
+                self._queryable_sobjects.append(sobject["name"].lower())
+
+        return self._queryable_sobjects
+
+    async def queryable_sobject_fields(self):
+        """Cached async property"""
+        if self._queryable_sobject_fields is not None:
+            return self._queryable_sobject_fields
+
+        self._queryable_sobject_fields = {}
+
+        for sobject in RELEVANT_SOBJECTS:
+            endpoint = DESCRIBE_SOBJECT_ENDPOINT.replace("<sobject>", sobject)
+            response = await self._get_json(f"{self.base_url}{endpoint}")
+
+            queryable_fields = [
+                f["name"].lower()
+                for f in response.get("fields", [])
+                if f["name"] in RELEVANT_SOBJECT_FIELDS
+            ]
+            self._queryable_sobject_fields[sobject] = queryable_fields
+
+        return self._queryable_sobject_fields
+
+    async def sobjects_cache_by_type(self):
+        """Cached async property
+
+        Many sobjects require extra data that is taxing on the rate limiter
+        to repeatedly fetch each request.
+        Instead we cache them on the first request for re-use later.
+        """
+        if self._sobjects_cache_by_type is not None:
+            return self._sobjects_cache_by_type
+
+        self._sobjects_cache_by_type = {}
+        self._sobjects_cache_by_type["Account"] = await self._prepare_sobject_cache(
+            "Account"
```

Comment:
The data fetched is from a different endpoint. It made sense in a structural sense to me to include the `Account` cache here instead of from the results of `get_account`. This is where all caching of SObjects is handled. It's easier to mentally track what's going on and is only 1 extra request.